src/misc.ts

```diff
@@ -0,0 +1,94 @@
+//
+// Types
+//
+
+/**
+ * Makes every specified property of the specified object type mutable.
+ *
+ * @template ObjectValue - The object whose readonly properties to make mutable.
+ * @template TargetKey - The property key(s) to make mutable.
+ */
+export type Mutable<
+  ObjectValue extends Record<string, unknown>,
+  TargetKey extends keyof ObjectValue
+> = {
+  -readonly [Key in keyof Pick<ObjectValue, TargetKey>]: ObjectValue[Key];
+} &
+  {
+    [Key in keyof Omit<ObjectValue, TargetKey>]: ObjectValue[Key];
+  };
+
+/**
+ * Useful for representing some value that _might_ be present and / or complete.
+ *
+ * @template Value - The value that might be present or complete.
+ */
+export type Maybe<Value> = Partial<Value> | null | undefined;
```

Comment:
How much freedom do we have currently in renaming things? I am wondering whether it makes sense to rename this type. `Maybe` is a pre-existing concept in functional programming to mean something that is there or not there. The `null | undefined` check here makes sense, but the `Partial<Value>` sticks out to me. 

Maybe `PartialOrNone` would be a better name?

Reply:
Oh we can rename anything! I like `PartialOrAbsent` :+1: